src/celeritas/track/detail/TrackInitAlgorithms.cu

```diff
@@ -74,7 +154,43 @@ size_type exclusive_scan_counts(
     CELER_DEVICE_API_CALL(PeekAtLastError());
 
     // Copy the last element (accumulated total) back to host
-    return ItemCopier<size_type>{stream_id}(stop.get() - 1);
+    auto result = ItemCopier<size_type>{stream_id}(stop.get() - 1);
+
+    // CUDA/HIP functions expect a cudaStream_t pointer for the stream
+    using StreamT = CELER_DEVICE_API_SYMBOL(Stream_t);
+    StreamT stream = device().stream(stream_id).get();
+    CELER_DEVICE_API_CALL(StreamSynchronize(stream));
```

Comment:
To simplify things a bit, you could add above line 146:
```c++
auto stream = device().stream(stream_id);
```
Then this becomes:
```suggestion
    stream.sync();
```
and similarly below you can use `stream.get()` and `stream.sync()` where needed.

Reply:
Sounds good. Updated with this change.